### PR TITLE
Feature/preprocessing for embedding component

### PR DESCRIPTION
**Pull request by Adem-Khachnaoui**
- textblocks are now pre-processed before being embedded.
- No code duplication between embedding and segmentation

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -11,6 +11,8 @@ services:
     ports:
       - 8000:8000
     working_dir: /usr/src/app
+    volumes:
+      - ./text_preprocessing:/usr/src/app/src/text_preprocessing
```
Comment (jpbernius) [inline docker-compose.yml:15]: I do not think this is enough. The Docker Containers will no longer work without the source code being available.  Can we copy it into the Docker Image? Maybe using a symlink?

File: docker-compose.yml [hunk 2]
```diff
@@ -24,6 +26,8 @@ services:
     ports:
       - 8001:8000
     working_dir: /usr/src/app
+    volumes:
+      - ./text_preprocessing:/usr/src/app/src/text_preprocessing
```
Comment (jpbernius) [inline docker-compose.yml:30]: I do not think this is enough. The Docker Containers will no longer work without the source code being available.  Can we copy it into the Docker Image? Maybe using a symlink?

Comment (Adem-Khachnaoui) [reply docker-compose.yml:30]: Sorry I don't get it.
I thought the volume is copied at start-up time so no need for the source to be available on runtime

Comment (jpbernius) [reply docker-compose.yml:30]: Volumes are mounted live into the container (which es great e.g. for development), but not copied.

Comment (linusmichel) [reply docker-compose.yml:30]: @Adem-Khachnaoui that would mean the container is not able to run "standalone" without the volume, hence without the repository. But on our processing-nodes, we won't have the repo cloned on there to run the containers. So we need to somehow make sure the ./text_preprocessing directory is copied to the container at buildtime, not only at runtime.

Comment (Adem-Khachnaoui) [reply docker-compose.yml]: Problem solved! please have a look

**Comment by Adem-Khachnaoui** [comment]
This PR is again ready for review. Please have a look !

**Review by linusmichel — DISMISSED** [review]
I ran into an issue with the new build context, if mongodb data exists in the root directory. Including
`data`
in the `.dockerignore` file resolves this. I added this change to the branch.

Rest looks good to me and works!

Maybe a few more comments in the code of the "benchmark"-stuff would help :)

**Comment by Adem-Khachnaoui** [comment]
> I ran into an issue with the new build context, if mongodb data exists in the root directory. Including
> `data`
> in the `.dockerignore` file resolves this. I added this change to the branch.
> 
> Rest looks good to me and works!
> 
> Maybe a few more comments in the code of the "benchmark"-stuff would help :)

 @linusmichel :
Thanks for the Review, 
I added some comments and a readme in the benchmark (as part of https://github.com/ls1intum/Athene/pull/24)

**Review by CanArisan — DISMISSED** [review]

**Review by jpbernius — DISMISSED** [review]

**Pull request merged**